kura/org.eclipse.kura.camel.cloud.factory/src/org/eclipse/kura/camel/cloud/factory/internal/CamelManager.java

```diff
@@ -147,15 +146,16 @@ private static String asString(Object object) {
 
     @Override
     public ComponentConfiguration getConfiguration() throws KuraException {
-        return new ComponentConfiguration() {
+        // FIXME: replace with ComponentConfiguration instance once issue #590 is fixed
+        return new ComponentConfigurationImpl() {
 
             @Override
             public String getPid() {
```

Comment:
@ctron Just out of curiosity, why are you not using the constructor, but instead using an anonymous class?
Thanks